musig/src/signer.rs

```diff
@@ -0,0 +1,258 @@
+use bellman::pairing::ff::Field;
+use franklin_crypto::jubjub::{JubjubEngine, FixedGenerators, Unknown};
+use franklin_crypto::eddsa::{PrivateKey, PublicKey, Signature, Seed};
+use franklin_crypto::jubjub::edwards::Point;
+use rand::{Rng, Rand};
+use franklin_crypto::rescue::RescueEngine;
+use crate::errors::MusigError;
+use crate::jubjub::JubJubWrapper;
+use crate::aggregated_pubkey::AggregatedPublicKey;
+use crate::hasher::Hasher;
+use crate::verifier::MuSigVerifier;
+
+pub struct MuSigSigner<E: JubjubEngine + RescueEngine>{
```

Comment:
I believe doc-comments (ideally, with doc-tests as examples) will be really helpful for users.

Also, this will make crate look nice once we publish it and its documentation will appear on `docs.rs`.